src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.ActionStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final ActionStrategy actionStrategy;
+
+    public ShopServiceImpl(ActionStrategy operationStrategy) {
+        this.actionStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            ActionHandler handler = actionStrategy.getHandler(transaction.getAction());
+            handler.apply(transaction.getFruit(), transaction.getValue());
```

Comment:
Ensure that the getHandler method in the ActionStrategy class does not violate the SOLID principles by making the ShopService dependent on other services. It should be independent, and the main() method should orchestrate the interactions between services.